src/utils.rs

```diff
@@ -20,20 +20,18 @@ pub fn is_aligned(shards: &[&[u8]]) -> bool {
             .all(|shard| shard.as_ptr().align_offset(ALIGN) != 0)
 }
 
-pub fn alloc_aligned<const SHARD_SIZE: usize>(shards: usize) -> Result<Vec<[u8; SHARD_SIZE]>> {
-    if SHARD_SIZE % ALIGN != 0 {
-        return Err(LeopardError::InvalidShardSize(SHARD_SIZE));
+pub fn alloc_aligned(shards: usize, shard_size: usize) -> Result<Vec<u8>> {
+    if shard_size % ALIGN != 0 {
+        return Err(LeopardError::InvalidShardSize(shard_size));
     }
 
-    let ratio = SHARD_SIZE / ALIGN;
-
     // create a vec of Aligned to make sure they are aligned properly
-    let aligned_mem = vec![Aligned::zeroed(); shards * ratio];
+    let aligned_mem = vec![Aligned::zeroed(); shards * shard_size / ALIGN];
```

Comment:
Any reasons on why you did't use [Layout::from_size_align](https://doc.rust-lang.org/stable/std/alloc/struct.Layout.html#method.from_size_align) + [alloc](std::alloc::alloc)

Reply:
likely just because I wasn't aware of it :+1:

Reply:
so this as well as prev attempt turned out to be unsound, violating rules in `Vec::from_raw_parts`. refactored whole thing